core/src/main/scala/org/http4s/Header.scala

```diff
@@ -18,118 +18,134 @@
  */
 package org.http4s
 
-import cats._
-import cats.data.NonEmptyList
-import cats.implicits.{catsSyntaxEither => _, _}
-import org.http4s.syntax.string._
+// import cats._
+// import cats.data.NonEmptyList
+// import cats.implicits.{catsSyntaxEither => _, _}
+// import org.http4s.syntax.string._
 import org.http4s.util._
-import scala.util.hashing.MurmurHash3
-
-/**
-  * Abstract representation o the HTTP header
-  * @see org.http4s.HeaderKey
-  */
-sealed trait Header extends Renderable with Product {
-  import Header.Raw
+// import scala.util.hashing.MurmurHash3
 
+trait Header[A] {
   def name: CaseInsensitiveString
+  def parse(s: String): Option[A]
+    // some headers like Set-Cookie may emit multiple actual headers
+  def toRawHeaders(a: A): List[Header.RawHeader]
```

Comment:
This flattening is to support `Headers#put`?  What does this do on a comma-separated header for headers that play by the rules?